### DISCUSSION TITLE
Help with configuration

### DISCUSSION THREAD

**Discussion by zigad** [rabbitmq/rabbitmq-perf-test#440] [Q&A]
Hi,

I have a question regarding on how to configure the test to run.

I have a RabbitMQ instance and Quarkus application that recieves the event from example "aaa" queue, process it and returns it back to example "bbb" queue. I have predefined exchanges "xxx" and "yyy", binding are made from "xxx" to "aaa" and from "yyy" to "bbb", so i will be using —predeclared option and that is fine.

The question now is how can I configure perf test so that producer will send messages to "xxx" exchange and consume messages from "bbb" queue. I have already setup so messages sent to "xxx" exchange are correct inc headers, but the problem I find is that I cannot specify on what queue the pert test will be consuming processed messages.

Overview of the message flow:

Producer sending to "xxx" exchange -> "xxx" exchange -> bindings route message to "aaa" queue -> Quarkus app -> "yyy" exchange -> bindings route message to "bbb" queue -> Consume and test performance from "bbb" queue.

I tried with something like this but it's not working as expected.
java -jar perf-test-2.18.0.jar --uri "customUri" --producers 1 --consumers 1 —predeclared -e "xxx" -k "customRoutingKey" -mp "customHeaders" --body body.json --predeclared -u "bbb" -z 60

**Comment by acogoluegnes** [comment]
It may be simpler to start 2 PerfTest instances, one to publish only (`-x 1 -y 0`) and one to consume only (`-x 0 -y 1`).

**Reply by zigad** [reply]
> It may be simpler to start 2 PerfTest instances, one to publish only (`-x 1 -y 0`) and one to consume only (`-x 0 -y 1`).

Hey, yes I thought about that but I would like to run the performance test to see how well my Quarkus application is performing, not the RabbitMQ instance. I think that if I do it like this I will get metrics for just consumption/production and not throughput of the Quarkus app :/

**Reply by acogoluegnes** [reply]
Have you tried to use the `-u` flag to specify the pre-declared queue you want PerfTest to consume from? Note PerfTest expects a specific message content to extract a timestamp and measure latency. You'll have to send back a message that contains the timestamp in the place PerfTest expects.

It may be a bit tricky, see the PerfTest consumer code that extracts the message timestamp to calculate latency (there's an option to use a header to store the timestamp):

https://github.com/rabbitmq/rabbitmq-perf-test/blob/4f3edd13673ecb59eabf8ce42ad264f14c7dee81/src/main/java/com/rabbitmq/perf/Consumer.java#L138-L154

Routing keys also need to match:

https://github.com/rabbitmq/rabbitmq-perf-test/blob/4f3edd13673ecb59eabf8ce42ad264f14c7dee81/src/main/java/com/rabbitmq/perf/Consumer.java#L281-L284

I've never done it but I think it's possible.

**Reply by zigad** [reply]
Thanks for this, I'll try it or maybe create a fork and try to modify it to meet my needs.
